AdminServer/appscale/admin/__init__.py

```diff
@@ -159,6 +220,46 @@ def initialize(self, acc, ua_client, zk_client, version_update_lock,
     self.version_update_lock = version_update_lock
     self.thread_pool = thread_pool
 
+  @gen.coroutine
+  def get(self, project_id, service_id):
+    """ Retrieves service versions.
+
+    Args:
+      project_id: A string specifying a project ID.
+      service_id: A string specifying a service ID.
+    Returns:
+      A dist in which each key is the service's version and
+      value is a dict of ports on which this version is deployed.
+    """
+    self.authenticate()
+
+    url = ('/appscale/projects/{project_id}/services/{service_id}/versions'
+           .format(project_id=project_id, service_id=service_id))
+    try:
+      versions = self.zk_client.get_children(url)
+    except NoNodeError:
+      message = 'Versions not found'
+      logging.warn('{}. URL: {}'.format(message, url))
+      raise CustomHTTPError(HTTPCodes.NOT_FOUND, message=message)
+
+    versions_info = {}
+    for version in versions:
+      versions_info[version] = ports_info = {}
+      try:
+        version_url = '{versions_url}/{version}'.format(
+          versions_url=url, version=version)
+        ports = json.loads(self.zk_client.get(version_url)[0])
+      except NoNodeError:
+        message = 'Information has not obtained for version: {}'.format(version)
+        logging.warn(message)
+        continue
+
+      extensions = ports['appscaleExtensions']
+      ports_info['http_port'] = extensions['httpPort']
+      ports_info['https_port'] = extensions['httpsPort']
+
+    self.finish(json_encode(versions_info))
```

Comment:
I think https port is not redundant information, e.g. if user wants to relocate version he should to specify both ports, or if he wants to update version's https port he should to have easy way to see which https port module's version is using now, doesn't it?